appwindow.js

```diff
@@ -52,6 +52,11 @@ var AppWindow = GObject.registerClass(
 
             this.toggle_action = this.simple_action('toggle', this.toggle.bind(this));
             this.hide_action = this.simple_action('hide', () => this.hide());
+            this.simple_action('toggle-maximize', () => {
+                this.settings.set_boolean('window-maximize', !this.settings.get_boolean('window-maximize'));
+            });
+            this.settings.set_boolean('window-maximize', false);
```

Comment:
Why is this (`set_boolean(..., false)`) necessary? What's wrong with keeping maximized state from the previous run? Remember, the application shuts down when you close all tabs. With this code, if you closed all tabs, and then opened the terminal again, it will be unmaximized

Reply:
On a new run my expectation is to return to "vanilla" state. I don't have a very strong preference (maybe it's something for the future, adding an option to control this behaviour)